### ISSUE TITLE
Serialization error

### ISSUE THREAD

**Issue by scmanjarrez** [piotrszyma/mcl-python#7]
Hi,

I encountered an intermittent error while testing the Python bindings for the MCL library. I haven't observed this issue with the C library, and I am unsure where to begin looking for a fix.

I tested both MCL 2.0 and 1.93, and the error occurred with both versions.
```
❯ python -m unittest tests/test_fp.py
......s...F.s....
======================================================================
FAIL: testSerialization (tests.test_fp.FpTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/s/mcl-python/tests/test_fp.py", line 47, in testSerialization
    self.assertEqual(fp, fp2)
AssertionError: <clas[21 chars]Fp'> 24072700270561460537064131549515187746788[68 chars]00595 != <clas[21 chars]Fp'> 0

----------------------------------------------------------------------
Ran 17 tests in 0.002s
```

The issue seems to occur intermittently during serialization. Occasionally, the pairing generated serializes to a weird value. The C function reports a return length of 48, but the bytes buffer in Python does not match this size. When deserialization is attempted, it fails, and the second object's getStr() returns 0.

```
Fp1:  <class '__main__.Fp'> 3019543027836134998822211675811522205718403523778478061120185809425329410468753782045394899768640645273331986043000
C serialize length: 48 | Python buffer len: 19
Fp1 serialize:  b"x\x90\x0fEnB\x90\xa5\x96r\xddJ\x0b'+\x06\x95\xdd1"
Fp2 deserialize Fp1 output
Fp2:  <class '__main__.Fp'> 0
C length: 48 | Python len: 0
Fp2 serialize:  b''
Fp1 == Fp2:  False
```

Any guidance on how to address or debug this issue would be appreciated.

Thank you!

**Comment by scmanjarrez** [comment]
Ok, I've been researching a bit and this seems to be a problem in the different types being used in the binding. You're using `ctypes.c_char * (BUFFER_SIZE + 1)` during serialization (returned buffer is treated as string), however, sometimes this is a void* in C, and sometimes null characters show up in this buffer:
I've been doing some research, and the issue seems to stem from differences in the types being used in the Python binding. Specifically, during serialization, you're using ctypes.c_char * (BUFFER_SIZE + 1) (which treats the returned buffer as a string). However, in the C library, this is a void*; null characters occasionally appear in the buffer, leading to the buffer being cut at this point in python.
```
3affffffa03515c26587b1653ffffffcf715453fffffff8ffffff8cffffffff72ffffffc5ffffff887-->00<--b67fffffffb79164665505a1efffffff1cffffff9cfffffff0ffffffa5ffffffc4fffffff441c61ffffff8affffffbdfffffffa52ffffffbcc
```
Yielding to the incorrect result in python, a serialized object way shorter than the actual one.

**Comment by scmanjarrez** [comment]
Fixed in #8
